src/platform/silabs/efr32/rs911x/wfx_sl_ble_init.c

```diff
@@ -0,0 +1,438 @@
+ * @param[in]  serv_handler, new service handler.
+ * @param[in]  handle, characteristic value attribute handle.
+ * @param[in]  att_type_uuid, attribute uuid value.
+ * @param[in]  val_prop, characteristic value property.
+ * @param[in]  data, characteristic value data pointer.
+ * @param[in]  data_len, characteristic value length.
+ * @return     none.
+ * @section description
+ * This function is used at application to create new service.
+ */
+
+void rsi_ble_add_char_val_att(void * serv_handler, uint16_t handle, uuid_t att_type_uuid, uint8_t val_prop, uint8_t * data,
+                                     uint8_t data_len, uint8_t auth_read)
+{
+    rsi_ble_req_add_att_t new_att = { 0 };
+
+    memset(&new_att, 0, sizeof(rsi_ble_req_add_att_t));
+    //! preparing the attributes
+    new_att.serv_handler  = serv_handler;
+    new_att.handle        = handle;
+    new_att.config_bitmap = auth_read;
+    memcpy(&new_att.att_uuid, &att_type_uuid, sizeof(uuid_t));
+    new_att.property = val_prop;
+
+    if (data != NULL)
+        memcpy(new_att.data, data, RSI_MIN(sizeof(new_att.data), data_len));
+
+    //! preparing the attribute value
+    new_att.data_len = data_len;
+
+    //! add attribute to the service
+    rsi_ble_add_attribute(&new_att);
+
+    if ((auth_read == ATT_REC_MAINTAIN_IN_HOST) || (data_len > 20))
+    {
+        if (data != NULL)
+        {
+            rsi_gatt_add_attribute_to_list(&att_list, handle, data_len, data, att_type_uuid, val_prop);
+        }
+    }
+
+    //! check the attribute property with notification/Indication
+    if ((val_prop & RSI_BLE_ATT_PROPERTY_NOTIFY) || (val_prop & RSI_BLE_ATT_PROPERTY_INDICATE))
+    {
+        //! if notification/indication property supports then we need to add client characteristic service.
+
+        //! preparing the client characteristic attribute & values
+        memset(&new_att, 0, sizeof(rsi_ble_req_add_att_t));
+        new_att.serv_handler       = serv_handler;
+        new_att.handle             = handle + 1;
+        new_att.att_uuid.size      = 2;
+        new_att.att_uuid.val.val16 = RSI_BLE_CLIENT_CHAR_UUID;
+        new_att.property           = RSI_BLE_ATT_PROPERTY_READ | RSI_BLE_ATT_PROPERTY_WRITE;
+        new_att.data_len           = 2;
+
+        //! add attribute to the service
+        rsi_ble_add_attribute(&new_att);
+    }
+
+    return;
+}
+
+
+/*==============================================*/
+/**
+ * @fn         rsi_ble_add_matter_service
+ * @brief      this function is used to add service for matter
+ * @return     status (uint32_t) 0 for success.
+ * @section description
+ * This function is used at application to create new service.
+ */
+
+uint32_t rsi_ble_add_matter_service(void)
+{
+    uuid_t custom_service    = { RSI_BLE_MATTER_CUSTOM_SERVICE_UUID };
+    custom_service.size      = RSI_BLE_MATTER_CUSTOM_SERVICE_SIZE;
+    custom_service.val.val16 = RSI_BLE_MATTER_CUSTOM_SERVICE_VALUE_16;

```

Comment:
What is this service exactly ?

Reply:
@jmeg-sfy This custom GATT service has been created for data transfer. RS911x module maintains the GATT service on the host , so created in the application context.